pkg/measure-config/src/measurefs.go

```diff
@@ -0,0 +1,278 @@
+	return "", nil
+}
+
+func getFileMap() (map[string]FileInfo, error) {
+	files := make(map[string]FileInfo)
+
+	walk_err := filepath.Walk("/config",
+		func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+
+			if info.IsDir() {
+				return nil
+			}
+			// may mark file as excluded but we will measure presence/absence
+			files[path] = FileInfo{exist: true, measureContent: !isInExcludeList(path)}
+			return nil
+		})
+	if walk_err != nil {
+		return nil, walk_err
+	}
+
+	// for every file in both exclude and risky lists add entries so the list of files
+	// is always the same across all devices in the world
+	for _, file := range getExcludeList() {
+		_, found := files[file]
+		if !found {
+			files[file] = FileInfo{exist: false, measureContent: false}
+		}
+	}
+
+	for _, file := range getDangerousList() {
+		_, found := files[file]
+		if !found {
+			files[file] = FileInfo{exist: false, measureContent: false}
+		}
+	}
+
+	return files, nil
+}
+
+func getSortedFileList(files map[string]FileInfo) []string {
+	keys := make([]string, 0, len(files))
+	for k := range files {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+	return keys
+}
+
+func measureConfig(tpm io.ReadWriter, password string) error {
+	files, err := getFileMap()
+
+	if err != nil {
+		return fmt.Errorf("cannot get file list: %v", err)
+	}
+
+	//get sorted list of files. We must always go the same order
+	//otherwise we'll get different PCR value even with exactly the same
+	//file names and their content
+	fileNames := getSortedFileList(files)
+
+	for _, file := range fileNames {
+		info := files[file]
+		var event *TpmEvent = nil
+
+		if info.exist {
+			if info.measureContent {
+				event, err = measureFileContent(file, tpm, password)
+			} else {
+				event, err = measureFilePath(file, tpm, password, true)
+			}
+
+			if err != nil {
+				return fmt.Errorf("cannot measure %s: %v", file, err)
+			}
+		} else {
+			event, err = measureFilePath(file, tpm, password, false)
+			if err != nil {
+				return fmt.Errorf("cannot measure %s: %v", file, err)
+			}
+		}
+		if err != nil {
+			return fmt.Errorf("cannot measure %s: %v", file, err)
+		}
+		//Now we have a new value of PCR and an event
+		//TODO: add events to the event log
+		// for now just print our measurements to boot log
+		log.Printf("%s pcr:%s", event.data, hex.EncodeToString(event.pcr))
+	}
+	return nil
+}
+
+func readConfigPCR(tpm io.ReadWriter) ([]byte, error) {
+	pcr, err := tpm2.ReadPCR(tpm, ConfigPCRIndex, tpm2.AlgSHA256)
+
+	if err != nil {
+		return nil, fmt.Errorf("cannot read PCR %d: %v", ConfigPCRIndex, err)
+	}
+	return pcr, nil
+}
+
+// Some file like generated certificates do not exist during the installation.
+// do we care? it seems nobody is using eve just after installation, but live image
+// won't report correct PCR
+func main() {
+	tpm, err := tpm2.OpenTPM(TpmDevicePath)
+	if err != nil {
+		log.Fatalf("couldn't open TPM device %s", TpmDevicePath)
+	}
```

Comment:
We run this container even on devices without a TPM.
In such a case it should be silent and not emit any fatal error.

Reply:
yes! fixed it